src/scripts/components/DropdownMultiSelectAdvanced.jsx

```diff
@@ -0,0 +1,246 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  LoaderPulse,
+  Input,
+  Button,
+  Icon,
+  Resource,
+  ResourceIntro,
+  ResourceGroup,
+  Scrollbars,
+  UtilitySystem,
+  Dropdown,
+  UtilityInlineGrid,
+  ResourceRight,
+} from '../components';
+
+class DropdownMultiSelectAdvanced extends React.Component {
+  state = {
+    searchText: '',
+    viewAllUsers: true,
+  }
+
+  sortMemberSearchIds = (a, b) => {
+    // Sort the rest by lastName
+    const nameA = this.props.selectedUsers[a].firstName.toLowerCase();
+    const nameB = this.props.selectedUsers[b].firstName.toLowerCase();
+
+    if (nameA < nameB) return -1;
+    if (nameA > nameB) return 1;
+
+    return 0;
+  }
+
+  handleUpdateSelectedIds = (id) => {
+    let selectedIds = this.props.selectedUserIds;
+    const { selectedUsers } = this.props;
+    const selectedUser = this.props.users[id];
+
+    const addAction = !selectedIds.includes(id);
+
+    if (addAction) {
+      selectedIds = selectedIds.concat(id);
+      selectedUsers[id] = selectedUser;
+    } else {
+      selectedIds = selectedIds.filter(selectedId => selectedId !== id);
+      delete selectedUsers[id];
+    }
+    if (selectedIds.length === 0) {
+      this.setState({ viewAllUsers: true });
+    }
+    this.props.handleUpdateSelectedIds(selectedIds, selectedUsers);
+  }
+
+  handleToggle = () => {
+    if (this.state.viewAllUsers) {
+      this.setState({ searchText: '' });
+      this.props.fetchUsersSearch('');
+    }
+    this.setState({ viewAllUsers: !this.state.viewAllUsers });
```

Comment:
Use prevState here for toggelling value of viewAllUsers.

Reply:
Hey @yogitattn and @mahimaag, one thing I noticed is we have member/user specific logic in here. I know that comes from the acceptance criteria. Is there a way to make this more generic to create a re-usable component? While we're only using this for a user dropdown at the moment, it's likely to be used in other cases moving forward. The component should be able to handle a variety of items and not be locked to members and last name sorting. The member-specific stuff should be passed to the component as context/props in rhinofront.

Let me know if there's anything I can do to help clarify. Thanks!

Reply:
@alexgsummer @benbruning what do y'all think of the approach here. I don't have enough context on this specific feature, but the original thought when I wrote the comment is that the component could be used for other lists besides users in the future (groups, channels, etc). 

If that's not the case then it is likely ok to keep as is, but if there is potential for re-use of the UI pattern, then I would recommend handling things like sorting logic in rhinofront and removing any references to users/members and replacing with list items.

Reply:
@juanfabrega Now I have amde some changes, removed the sorting logic and moved it to rhinofront side.

Reply:
I agree with @juanfabrega, while this may only currently be used for Users specifically right now, rhinostyle components should be generic and be able to be used for other types of data.  If it needs to be so specific to users, it probably shouldn't be a rhinostyle component.

Reply:
Agreed. This component shouldn't use `users` for any variable names and remain generic. 

Functions, like `formatAvatarName` probably shouldn't be in RhinoStyle. The consumer of this component will either need to pass in a formatter function or pass in data that can be consumed and displayed "as-is".

Reply:
@juanfabrega @alexgsummer @benbruning I have made all changes regarding making this component fully generic. Now It can be used for another type of list also, In case of a user, We just need to pass a prop 'type' as 'member' to render avatar, but for other types, only list will be displayed with titles only. I have also made  all variables and functions names generic without any 'user' keyword and have removed this 'formatAvatarName' function from rhinostyle . Please review once.